### PR TITLE
[sample_vpp] Fixes regressions after recent refactoring 

### PR DESCRIPTION
**Pull request by AntonGrishin**
1) Wrong variable usage;
2) Change InitParamsVPP signature
3) When we use mfxFrameSurfaceWrap and pass them to common function, we lose information about free surfaces, so change source to template

### CODE REVIEW

File: samples/sample_common/include/sample_utils.h [hunk 1]
```diff
@@ -479,7 +479,7 @@ class ExtBufHolder : public T
             RefreshBuffers();
             return m_ext_buf.back();
         }
-
+        RefreshBuffers();
```
Comment (dmitryermilov) [inline samples/sample_common/include/sample_utils.h]: IMO, not needed here. Please reset class obj at app side.

File: samples/sample_vpp/src/sample_vpp_utils.cpp [hunk 2]
```diff
@@ -2011,6 +2011,22 @@ mfxStatus UpdateSurfacePool(mfxFrameInfo SurfacesInfo, mfxU16 nPoolSize, mfxFram
     return MFX_ERR_NONE;
 }
 
+mfxU16 GetFreeSurfaceIndex(mfxFrameSurfaceWrap* pSurfacesPool, mfxU16 nPoolSize)
```
Comment (dmitryermilov) [inline samples/sample_vpp/src/sample_vpp_utils.cpp]: please make in template method

File: samples/sample_vpp/src/sample_vpp_utils.cpp [hunk 3]
```diff
@@ -307,7 +307,7 @@ mfxStatus ParseGUID(msdk_char strPlgGuid[MSDK_MAX_FILENAME_LEN], mfxU8 DataGUID[
     return MFX_ERR_NONE;
 }
 
-mfxStatus InitParamsVPP(mfxVideoParam* pParams, sInputParams* pInParams, mfxU32 paramID)
+mfxStatus InitParamsVPP(MfxVideoParamsWrapper* pParams, sInputParams* pInParams, mfxU32 paramID)
```
Comment (dmitryermilov) [inline samples/sample_vpp/src/sample_vpp_utils.cpp]: need "*pParams = {}" to reset

File: samples/sample_vpp/src/sample_vpp_utils.cpp (hunk 2 above)
Comment (AntonGrishin) [reply samples/sample_vpp/src/sample_vpp_utils.cpp]: Done

File: samples/sample_common/include/sample_utils.h [hunk 4]
```diff
@@ -992,7 +992,30 @@ class CAutoTimer
...
+#ifndef MSDK_INVALID_SURF_IDX
+    #define MSDK_INVALID_SURF_IDX 0xFFFF
+#endif
+
+template <class T>
+mfxU16 GetFreeSurfaceIndex(T* pSurfacesPool, mfxU16 nPoolSize)
+{
+    if (pSurfacesPool)
+    {
+        for (mfxU16 i = 0; i < nPoolSize; i++)
```
Comment (mgonchar) [inline samples/sample_common/include/sample_utils.h]: use ```std::find_if```

File: samples/sample_common/include/sample_utils.h [hunk 5]
```diff
@@ -992,7 +992,30 @@ class CAutoTimer
 
 mfxStatus ConvertFrameRate(mfxF64 dFrameRate, mfxU32* pnFrameRateExtN, mfxU32* pnFrameRateExtD);
 mfxF64 CalculateFrameRate(mfxU32 nFrameRateExtN, mfxU32 nFrameRateExtD);
-mfxU16 GetFreeSurfaceIndex(mfxFrameSurface1* pSurfacesPool, mfxU16 nPoolSize);
+
+#ifndef MSDK_INVALID_SURF_IDX
+    #define MSDK_INVALID_SURF_IDX 0xFFFF
```
Comment (mgonchar) [inline samples/sample_common/include/sample_utils.h]: make it ```constexpr``` and move to function's body:
```
template <class T>
mfxU16 GetFreeSurfaceIndex(T* pSurfacesPool, mfxU16 nPoolSize)
{
    constexpr mfxU16 MSDK_INVALID_SURF_IDX = 0xffff;
```

Comment (AntonGrishin) [reply samples/sample_common/include/sample_utils.h]: done

File: samples/sample_common/include/sample_utils.h (hunk 4 above)
Comment (AntonGrishin) [reply samples/sample_common/include/sample_utils.h]: i think we have more code in this case 
```
mfxU16 i = 0;
auto it = std::find_if(pSurfacesPool, pSurfacesPool + nPoolSize, [&i](const T & surface)
{
    i++;
    return (0 == surface.Data.Locked);
});

if (it != (pSurfacesPool + nPoolSize))
    return i-1;
```

Comment (dmitryermilov) [reply samples/sample_common/include/sample_utils.h]: @mgonchar , please comment. We need to address the regression ASAP

Comment (mgonchar) [reply samples/sample_common/include/sample_utils.h]: I'm ok to merge it as is.

I thought about different variant, but in general I agree with you
```
auto it = std::find_if(pSurfacesPool, pSurfacesPool + nPoolSize, [](const T & surface)
{    return 0 == surface.Data.Locked; });
if (it == pSurfacesPool + nPoolSize)
   return MSDK_INVALID_SURF_IDX;

return it - pSurfacesPool;
```

**Pull request merged**